x-pack/plugins/security/public/nav_control/nav_control_service.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Subscription } from 'rxjs';
+import { CoreStart } from 'src/core/public';
+import ReactDOM from 'react-dom';
+import React from 'react';
+import { SecurityLicense } from '../../common/licensing';
+import { AuthenticatedUser } from '../../common/model';
+import { SecurityNavControl } from './nav_control_component';
+
+interface SetupDeps {
+  securityLicense: SecurityLicense;
+}
+
+interface StartDeps {
+  core: CoreStart;
+}
+
+export class SecurityNavControlService {
+  private securityLicense!: SecurityLicense;
+
+  private navControlRegistered!: boolean;
+
+  private securityFeaturesSubscription?: Subscription;
+
+  public setup({ securityLicense }: SetupDeps) {
+    this.securityLicense = securityLicense;
+  }
+
+  public start({ core }: StartDeps) {
+    this.securityFeaturesSubscription = this.securityLicense.features$.subscribe(
+      ({ showLinks }) => {
+        const isAnonymousPath = core.http.anonymousPaths.isAnonymous(window.location.pathname);
+
+        const shouldRegisterNavControl =
+          !isAnonymousPath && showLinks && !this.navControlRegistered;
+
+        if (shouldRegisterNavControl) {
+          const user = core.http.get('/api/security/v1/me', {
```

Comment:
Another PR has deprecated this route, and we should now be using `/internal/security/me`

Reply:
Good catch, I'll update this